<?xml version="1.0"?>
<launch>
    <param name="strategy" value="CYCLE"/>

    <node pkg="patrol_sim" type="monitor.py" name="monitor" output="screen" cwd="node">
        <param name="nb_drones" value="6"/>
        <param name="nb_copter" value="0"/>
        <param name="nb_rover" value="6"/>
    </node>
    <group ns="drone0">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5760/?ids=1,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4243@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="1"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone0"/>
            <param name="instance" value="0"/>
        </node>
    </group>
    <group ns="drone1">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5770/?ids=2,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4244@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="2"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone1"/>
            <param name="instance" value="1"/>
        </node>
    </group>
    <group ns="drone2">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5780/?ids=3,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4245@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="3"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone2"/>
            <param name="instance" value="2"/>
        </node>
    </group>
    <group ns="drone3">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5790/?ids=4,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4246@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="4"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone3"/>
            <param name="instance" value="3"/>
        </node>
    </group>
    <group ns="drone4">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5800/?ids=5,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4247@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="5"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone4"/>
            <param name="instance" value="4"/>
        </node>
    </group>
    <group ns="drone5">
        <!--<arg name="fcu_url" default="udp://127.0.0.1:14550@14555"/>-->
        <arg name="fcu_url" default="tcp://127.0.0.1:5810/?ids=6,240"/>
        <arg name="gcs_url" default="udp-pb://127.0.0.1:4248@4242"/>
        <!--<arg name="gcs_url" default=""/>-->
        <arg name="tgt_system" default="6"/>
        <arg name="tgt_component" default="1"/>
        <arg name="log_output" default="screen"/>

        <include file="$(find mavros)/launch/node.launch">
            <arg name="pluginlists_yaml" value="$(find patrol_sim)/launch/apm_pluginlists.yaml"/>
            <arg name="config_yaml" value="$(find patrol_sim)/launch/apm_config.yaml"/>

            <arg name="fcu_url" value="$(arg fcu_url)"/>
            <arg name="gcs_url" value="$(arg gcs_url)"/>
            <arg name="tgt_system" value="$(arg tgt_system)"/>
            <arg name="tgt_component" value="$(arg tgt_component)"/>
            <arg name="log_output" value="$(arg log_output)"/>
        </include>
        <node pkg="patrol_sim" type="patroler.py" name="patroler" output="screen">
            <param name="vehicle_name" value="drone5"/>
            <param name="instance" value="5"/>
        </node>
    </group>

</launch>

